.github/instructions/test-scenario-prompt.instructions.md

```diff
@@ -0,0 +1,569 @@
+
+❌ **Don't be vague**: Avoid instructions like "test the feature" without specifics.
+
+❌ **Don't assume context**: Don't assume the agent knows about PRs, issues, or features being tested.
+
+❌ **Don't skip verification**: Every action should have a verification step.
+
+❌ **Don't make scenarios too narrow**: Scenarios should test meaningful workflows, not single function calls.
+
+❌ **Don't require manual intervention**: Scenarios must be fully automatable.
+
+❌ **Don't test cloud-only services**: Avoid scenarios requiring paid subscriptions or cloud credentials.
+
+❌ **Don't duplicate unit test coverage**: Scenarios are for integration testing, not unit testing.
+
+❌ **Don't create scenarios without request**: Never automatically create scenarios—only suggest them.
+
+## Testing Your Scenario
+
+After creating a scenario, test it by:
+
+1. **Commit the prompt.md file** to your PR branch
+2. **Comment on the PR**: `/test-scenario your-scenario-name`
+3. **Monitor the workflow**: Check the workflow run in the Actions tab
+4. **Review the created issue**: Follow the link to see the agent's work in aspire-playground
+5. **Iterate if needed**: Update the prompt based on results and test again
+
+## Common Patterns
+
+### Pattern: CLI Installation
+
+Most scenarios need to install the Aspire CLI from the PR build:
+
+```markdown
+## Step 1: Install the Aspire CLI from the PR Build
+
+The aspire-playground repository includes comprehensive instructions for acquiring 
+different versions of the CLI, including PR builds.
+
+**Follow the CLI acquisition instructions already provided in the aspire-playground 
+repository to obtain the native AOT build of the CLI for this PR.**
+
+Once acquired, verify the CLI is installed correctly:
+
+\```bash
+aspire --version
+\```
+
+Expected output should show the version matching the PR build.
+```
+
+### Pattern: Creating an Application
+
+Standard pattern for creating a new Aspire application:
+
+```markdown
+## Step 2: Create a New Aspire Application
+
+Use `aspire new` to create a new application:
+
+\```bash
+aspire new
+\```
+
+Follow the interactive prompts to select the desired template and options.
+
+Verify the project structure:
+
+\```bash
+ls -la
+\```
+
+Expected structure:
+- `AppName.sln` - Solution file
+- `AppName.AppHost/` - The Aspire AppHost project
+- Additional project directories based on template selection
+```
+
+### Pattern: Running and Verifying
+
+Standard pattern for running and verifying an application:
+
+```markdown
+## Step 3: Run the Application
+
+Start the application:
+
+\```bash
+aspire run
+\```
+
+Wait for startup (30-60 seconds) and note the Dashboard URL from the output.
+
+### 3.1 Verify the Dashboard
+
+Navigate to the Dashboard using the URL from the output:
+
+\```bash
+playwright-browser navigate $DASHBOARD_URL
+playwright-browser take_screenshot --filename dashboard.png
+\```
+
+Expected: Dashboard loads successfully and screenshot shows all resources running.
```

Comment:
Should this say that it might need to wait for the resources to either be ready or fail before taking the screenshot? And if it did fail then would any elements on the page need to be expanded to get useful information?
Would multiple screenshots be useful it is taking longer than some threshold?